SimplenoteTests/NoticeControllerTests.swift

```diff
@@ -0,0 +1,238 @@
+        XCTAssertNotEqual(notice.action?.title, presenter.lastNoticeView?.actionTitle)
+        XCTAssertNotNil(presenter.lastNoticeView?.handler)
+    }
+
+    func testDismiss() {
+        let noticeA = Notice(message: "Message A", action: nil)
+
+        controller.present(noticeA)
+        var expectedActions: [MockNoticePresenter.Action] = [
+            .present(noticeA.message)
+        ]
+
+        timerFactory.timer?.fire()
+        expectedActions.append(.dismiss(noticeA.message))
+
+        XCTAssertEqual(expectedActions, presenter.actionLog)
+    }
+
+    func testDismissContinuesToNextNotice() {
+        let noticeA = Notice(message: "Message A", action: nil)
+        let noticeB = Notice(message: "Message B", action: nil)
+
+        controller.present(noticeA)
+        controller.present(noticeB)
+
+        var expectedActions: [MockNoticePresenter.Action] = [
+            .present(noticeA.message)
+        ]
+        XCTAssertEqual(presenter.actionLog, expectedActions)
+
+        timerFactory.timer?.fire()
+
+        expectedActions.append(.dismiss(noticeA.message))
+        expectedActions.append(.present(noticeB.message))
+        XCTAssertEqual(presenter.actionLog, expectedActions)
+
+        timerFactory.timer?.fire()
+
+        expectedActions.append(.dismiss(noticeB.message))
+        XCTAssertEqual(presenter.actionLog, expectedActions)
+    }
+
+    func testPressingOnActionDismissesNotice() {
+        let action = NoticeAction(title: "Action", handler: {})
+        let noticeA = Notice(message: "Message A", action: action)
+
+        controller.present(noticeA)
+        var expectedActions: [MockNoticePresenter.Action] = [
+            .present(noticeA.message)
+        ]
+
+        controller.actionWasTapped()
+        expectedActions.append(.dismiss(noticeA.message))
+
+        XCTAssertEqual(expectedActions, presenter.actionLog)
+    }
+
+    func testLongPressInvalidatesTimerAndNoticeDoesNotDismiss() {
+        let noticeA = Notice(message: "Message A", action: nil)
+
+        controller.present(noticeA)
+        let expectedActions: [MockNoticePresenter.Action] = [
+            .present(noticeA.message)
+        ]
+
+        controller.noticePressBegan()
+
+        XCTAssertEqual(expectedActions, presenter.actionLog)
+        XCTAssertNil(timerFactory.timer?.completion)
+    }
+
+    func testLongPressReleasedDismissesTimer() {
+        let noticeA = Notice(message: "Message A", action: nil)
+
+        controller.present(noticeA)
+        var expectedActions: [MockNoticePresenter.Action] = [
+            .present(noticeA.message)
+        ]
+
+        controller.noticePressBegan()
+
+        XCTAssertEqual(expectedActions, presenter.actionLog)
+        XCTAssertNil(timerFactory.timer?.completion)
+
+        controller.noticePressEnded()
+        timerFactory.timer?.fire()
+        expectedActions.append(.dismiss(noticeA.message))
+
+        XCTAssertEqual(expectedActions, presenter.actionLog)
+    }
```

Comment:
Just a note: because we already tested some parts in other places, we can only assert things that we didn't check yet.
For example, because most of this method is tested in `testLongPressInvalidatesTimerAndNoticeDoesNotDismiss` it could look like:
```
func testLongPressReleasedDismissesTimer() {
    let noticeA = Notice(message: "Message A", action: nil)

    controller.present(noticeA)
    controller.noticePressBegan()
    controller.noticePressEnded()

    XCTUnwrap(timerFactory.timer).fire() <- can use XCTUnwrap to make sure we actually created a timer

    var expectedActions: [MockNoticePresenter.Action] = [
      .present(noticeA.message),
      .dismiss(noticeA.message)
    ]
    XCTAssertEqual(expectedActions, presenter.actionLog)
}
```